media/whip_server.go

```diff
@@ -86,24 +89,60 @@ func (s *WHIPServer) CreateWHIP(ctx context.Context, ssr *SwitchableSegmentReade
 	}
 	defer r.Body.Close()
 
+	// lock while we create the peer connection and wait to receive the stats getter via the statsInterceptorFactory
+	s.mu.Lock()
 	// Create a new PeerConnection
 	peerConnection, err := s.api.NewPeerConnection(WebrtcConfig)
 	if err != nil {
+		s.mu.Unlock()
 		clog.InfofErr(ctx, "Failed to create peerconnection", err)
 		http.Error(w, "Failed to create PeerConnection", http.StatusInternalServerError)
 		return NewMediaStateError(errors.New("failed to create peerconnection"))
 	}
+	statsGetter := <-s.statsGetters
+	s.mu.Unlock()
+
 	mediaState := NewMediaState(peerConnection)
 
 	// OnTrack callback: handle incoming media
 	trackCh := make(chan *webrtc.TrackRemote)
 	peerConnection.OnTrack(func(track *webrtc.TrackRemote, receiver *webrtc.RTPReceiver) {
-		clog.Info(ctx, "New track", "codec", track.Codec().MimeType, "ssrc", track.SSRC())
+		go func() {
+			// Periodically check whip stats and write logs and metrics
+			for !mediaState.IsClosed() {
+				trackStats := statsGetter.Get(uint32(track.SSRC()))
+				if trackStats == nil {
+					clog.Info(ctx, "whip trackStats was nil")
+				} else {
+					clog.Info(ctx, "whip InboundRTPStreamStats", "jitter", trackStats.InboundRTPStreamStats.Jitter, "packets_lost", trackStats.InboundRTPStreamStats.PacketsLost, "rtt", trackStats.RemoteInboundRTPStreamStats.RoundTripTime) // TODO more stats
+					// TODO prometheus metric for jitter, packets lost etc
+				}
+
+				statsReport := peerConnection.GetStats()
+
+				for _, stat := range statsReport {
```

Comment:
When we start serving audio, this would make us print the transport stats repeatedly for each track